### PR TITLE
yaml text upload method implemented.

### PR DESCRIPTION
**Pull request by kingshuk-nandy**

### CODE REVIEW

File: cypress/fixtures/deployment.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+apiVersion: apps/v1
```
Comment (jruddell) [inline cypress/fixtures/deployment.yml:1]: I'd rather not hardcode a yaml file here, we have a template we use and that has been changed a few times already.. don't want extra sources we need to maintain. can we reuse our template here?

**Review by jruddell — APPROVED** [review]

File: cypress/support/commands/selectorUtility.ts [hunk 2]
```diff
@@ -44,25 +45,38 @@ Cypress.Commands.add('assertMatchOrExist', (dataTestId: string, text?: string) =
 
 Cypress.Commands.add(
   'navigateToSpecificSection',
-  (mainSectionId: string, subSectionId: string, apiCalls?: string[]) => {
+  (mainSectionId: string, subSectionId: string, apiCalls?: string[], contentTestId?: string) => {
     cy.getByTestId(mainSectionId, 10000).click({ force: true })
 
-    if (apiCalls.includes('1')) {
+    if (apiCalls !== undefined && apiCalls.includes('1')) {
```
[the remark this answers is not in the document]
Comment (kingshuk-nandy) [reply cypress/support/commands/selectorUtility.ts:51]: Done

File: cypress/support/commands/selectorUtility.ts [hunk 3]
```diff
@@ -44,25 +45,38 @@ Cypress.Commands.add('assertMatchOrExist', (dataTestId: string, text?: string) =
...
     }
     cy.clickOnNavTab(subSectionId)
 
-    if (apiCalls.length !== 0) {
+    if (apiCalls !== undefined && apiCalls.length !== 0) {
       cy.callInterception(apiCalls)
     }
     cy.get('body', { timeout: 20000 }).then((body) => {
-      if (body.find('[data-testid="check-box-cluster-selection"]').length > 0) {
+      const id =
```
[the remark this answers is not in the document]
Comment (kingshuk-nandy) [reply cypress/support/commands/selectorUtility.ts:62]: here, contentTestId may be different for the different elements. If contentTestId is not provided then we assign the default contentTestId and check if the element of that id is present or not.

[the remark this answers is not in the document]
Comment (jruddell) [reply cypress/support/commands/selectorUtility.ts:62]: I understand that. What i mean is you don't need all of this code to do the same thing as 

`(mainSectionId: string, subSectionId: string, apiCalls?: string[], contentTestId = '[data-testid="check-box-cluster-selection"]') => {`

[the remark this answers is not in the document]
Comment (kingshuk-nandy) [reply cypress/support/commands/selectorUtility.ts]: done

**Review by jruddell — APPROVED** [review]

**Pull request merged**